db/view/view_update_generator.cc

```diff
@@ -33,28 +35,55 @@ future<> view_update_generator::start() {
             if (_sstables_with_tables.empty()) {
                 _pending_sstables.wait().get();
             }
-            while (!_sstables_with_tables.empty()) {
-                auto& [sst, t] = _sstables_with_tables.front();
+
+            // If we got here, we will process all tables we know about so far eventually so there
+            // is no starvation
+            for (auto& t : _sstables_with_tables | boost::adaptors::map_keys) {
+                schema_ptr s = t->schema();
+
+                // Copy what we have so far so we don't miss new updates
+                typename decltype(_sstables_with_tables)::mapped_type sstables = {};
+                std::swap(sstables, _sstables_with_tables[t]);
+
                 try {
-                    schema_ptr s = t->schema();
-                    flat_mutation_reader staging_sstable_reader = sst->read_rows_flat(s, no_reader_permit());
-                    auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, _db, sst, _as), db::no_timeout);
+                    // temporary: need an sstable set for the flat mutation reader, but the
+                    // compaction_descriptor takes a vector. Soon this will become a compaction
+                    // so the transformation to the SSTable set will not be needed.
+                    auto ssts = make_lw_shared(t->get_compaction_strategy().make_sstable_set(s));
+                    for (auto& sst : sstables) {
+                        ssts->insert(sst);
+                    }
+
+                    flat_mutation_reader staging_sstable_reader =
+                        ::make_range_sstable_reader(s,
```

Comment:
wild indent

Reply:
what's the problem with this ident ?

Reply:
Its wild. :laughing: 

Broken lines should be double indented. But in fact this line shouldn't even be broken.

Reply:
I broke the first line because it was reaching the end of my screen.

Reply:
I'll change it anyway because the only reason for this code to exist is to soon disappear when I change this to a compaction